fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/JoinedRecordTypeBuilder.java

```diff
@@ -122,7 +162,8 @@ public JoinedRecordTypeBuilder(@Nonnull RecordMetaDataProto.JoinedRecordType typ
             addConstituent(joinConstituent.getName(), metaDataBuilder.getRecordType(joinConstituent.getRecordType()), joinConstituent.getOuterJoined());
         }
         for (RecordMetaDataProto.JoinedRecordType.Join join : typeProto.getJoinsList()) {
-            addJoin(join.getLeft(), KeyExpression.fromProto(join.getLeftExpression()), join.getRight(), KeyExpression.fromProto(join.getRightExpression()));
+            addJoin(join.getLeft(), KeyExpression.fromProto(join.getLeftExpression()), KeyExpression.fromProto(join.getLeftJoinExpression()),
+                    join.getRight(), KeyExpression.fromProto(join.getRightExpression()), KeyExpression.fromProto(join.getRightJoinExpression()));
```

Comment:
I don't think this is backwards compatible. In particular, if an existing `JoinedRecordType` does not have the `left_join_expression` and `right_join_expression` set (because it was serialized before those fields existed), this will now give `KeyExpression.toProto` a default instance of the proto which will throw a `DeserializationException`. I believe the right behavior is to instead insert the `left_expression` and `right_expression` if the proto does not have the join expression set.

Reply:
fixed.